src/Frontend/actions/localeActions.js

```diff
@@ -26,14 +26,17 @@ import ApiKey             from '../util/http/ApiKey';
  *
  * @returns {Object} The payload for the reducers.
  */
-export const changeLocale = (locale) => {
-  //TODO: Fix auth
-  if (false) {
+export const changeLocale = (locale) => (dispatch, state) => {
+  const { authenticated, appProfile } = state().user.security;
+  if (authenticated) {
     axios.patch('/api/protected/locale.json', { locale }, {
-      headers: { 'X-API-KEY': ApiKey.getApiKey() }
+      headers: { 'X-API-KEY': appProfile.apiKey }
     });
   }
-  return ({
+
+  Locale.setLocale(locale);
+
+  dispatch({
```

Comment:
it's the way how `redux-thunk` work. You return another callback which has the `dispatch` and `getState()` prop. Needed to access the state since redux should be the single source of truth and I wanted to get rid of these crappy `Locale.getStuff()` calls (or is there a better approach? Asking since I don't think we should include the store everywhere)